src/kixi/nhs/board_report.clj

```diff
@@ -7,7 +7,58 @@
             [kixi.nhs.data.transform                       :as transform]
             [kixi.nhs.patient-experience.deprivation       :as deprivation]
             [kixi.nhs.patient-experience.ethnicity         :as ethnicity]
-            [kixi.nhs.patient-experience.gender-comparison :as gender]))
+            [kixi.nhs.patient-experience.gender-comparison :as gender]
+            [clj-time.format                               :as tf]
+            [clj-time.core                                 :as t]
+            [clj-time.coerce                               :as tc]))
+
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;; Idicator 65: Incidence of MRSA                                                       ;;
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+;; TODO this formatter will be changed when bug in CKAN is fixed and data re-uploaded
+(def custom-formatter (tf/formatter "yyyy-MM-dd'T'HH:mm:ss"))
+
+(defn latest-month
+  "Finds the latest month in data.
+  It looks for a date using provided key.
+  Returns a datetime."
+  [k data]
+  (->> data
+       (map #(tf/parse custom-formatter (k %)))
+       t/latest))
+
+(defn enrich
+  "Use latest month timestamp to enrich the map
+  with year and period_of_coverage which are PKs in
+  datastore.
+  Stringifies the sum value."
+  [timestamp m]
+  (-> m
+      (assoc :year (str (t/year timestamp))
+             :period_of_coverage (tf/unparse custom-formatter timestamp))
+      (update-in [:sum] str)))
+
+(defn incidence
+  "Reads data from CKAN for a given resource_id,
+  filters data for latest month, sums up values for all
+  CCGs."
+  [ckan-client recipe-map]
+  (let [data           (storage/get-resource-data ckan-client (:resource-id recipe-map))
+        timestamp      (latest-month :reporting_period data)
+        updated-recipe (update-in recipe-map [:conditions] conj {:field :reporting_period
+                                                                 :values #{(tf/unparse custom-formatter timestamp)}})]
+    (->> data
+         (transform/filter-dataset updated-recipe)
+         (transform/sum-sequence (:sum-field recipe-map)) ;; returns a single map
+         (enrich timestamp)
+         (conj []) ;; dataset should be a sequence of maps
+         (transform/enrich-dataset recipe-map)
```

Comment:
This feels like the function in enrich-data set (what the mapv calls) should be separate and callable from transform/